src/game/game.jsx

```diff
@@ -0,0 +1,304 @@
+		if (array[i + (counter * 2)] === aiSymbol && array[i + 1 + (counter * 2)] === aiSymbol && array[i + 2 + (counter * 2)] !== huSymbol) {
+			i = i + 2 + (counter * 2);
+			return i;
+		} else if (array[i + (counter * 2)] === aiSymbol && array[i + 2 + (counter * 2)] === aiSymbol && array[i + 1 + (counter * 2)] !== huSymbol) {
+			i= i + 1 + (counter * 2);
+			return i;
+		} else if (array[i + 1 + (counter * 2)] === aiSymbol && array[i + 2 + (counter * 2)] === aiSymbol && array[i + (counter * 2)] !== huSymbol) {
+			i = i + (counter * 2);
+			return i;
+		}
+		counter++;
+	}
+	
+	if (array[0] === aiSymbol && array[4] === aiSymbol && array[8] !== huSymbol) {
+		return 8;
+	} else if (array[0] === aiSymbol && array[8] === aiSymbol && array[4] !== huSymbol) {
+		return 4;
+	} else if (array[4] === aiSymbol && array[8] === aiSymbol && array[0] !== huSymbol) {
+		return 0;
+	} else if (array[2] === aiSymbol && array[4] === aiSymbol && array[6] !== huSymbol) {
+		return 6;
+	} else if (array[2] === aiSymbol && array[6] === aiSymbol && array[4] !== huSymbol) {
+		return 4;
+	} else if (array[6] === aiSymbol && array[4] === aiSymbol && array[2] !== huSymbol) {
+		return 2;
+	}
+	
+	for (let i = 0; i <= 2; i++) {
+		if (array[i] === huSymbol && array[i + 3] === huSymbol && array[i + 6] !== aiSymbol) {
+			return i + 6;
+		} else if (array[i] === huSymbol && array[i + 6] === huSymbol && array[i + 3] !== aiSymbol) {
+			return i + 3;
+		} else if (array[i + 3] === huSymbol && array[i + 6] === huSymbol && array[i] !== aiSymbol) {
+			return i;
+		}
+	}
+	counter = 0;
+	
+	for (let i = 0; i <= 2; i++) {
+		if (array[i + (counter * 2)] === huSymbol && array[i + 1 + (counter * 2)] === huSymbol && array[i + 2 + (counter * 2)] !== aiSymbol) {
+			i = i + 2 + (counter * 2);
+			return i;
+		} else if (array[i + (counter * 2)] === huSymbol && array[i + 2 + (counter * 2)] === huSymbol && array[i + 1 + (counter * 2)] !== aiSymbol) {
+			i= i + 1 + (counter * 2);
+			return i;
+		} else if (array[i + 1 + (counter * 2)] === huSymbol && array[i + 2 + (counter * 2)] === huSymbol && array[i + (counter * 2)] !== aiSymbol) {
+			i = i + (counter * 2);
+			return i;
+		}
+		counter++;
+	}
+	
+	if (array[0] === huSymbol && array[4] === huSymbol && array[8] !== aiSymbol) {
+		return 8;
+	} else if (array[0] === huSymbol && array[8] === huSymbol && array[4] !== aiSymbol) {
+		return 4;
+	} else if (array[4] === huSymbol && array[8] === huSymbol && array[0] !== aiSymbol) {
+		return 0;
+	} else if (array[2] === huSymbol && array[4] === huSymbol && array[6] !== aiSymbol) {
+		return 6;
+	} else if (array[2] === huSymbol && array[6] === huSymbol && array[4] !== aiSymbol) {
+		return 4;
+	} else if (array[6] === huSymbol && array[4] === huSymbol && array[2] !== aiSymbol) {
+		return 2;
+	}
+	
+	if (array[4] !== aiSymbol && array[4] !== huSymbol) {
+		return 4;
+	}
+	
+	for (let i = 0; i <= 8; i++) {
+		if (array[i] !== huSymbol && array[i] !== aiSymbol) {
+			return i;
+		}
+	}
```

Comment:
don't repeat the code, avoid repetitive operations

Reply:
omg! I did it :)) optimized from 91 lines to 32. How it looks? :))
```suggestion
function doAITurn (array, huSymbol, aiSymbol) {
	const moveMatrix = [
		[0, 1, 2],
		[3, 4, 5],
		[6, 7, 8],
		[0, 3, 6],
		[1, 4, 7],
		[2, 5, 8],
		[0, 4, 8],
		[2, 4, 6],
	];

	for (let i = 0; i < moveMatrix.length; i++) {
		const [a, b, c] = moveMatrix[i];
		if (array[a] === array[b] && (array[a] !== null && array[b] !== null) && array[c] === null) {
			return c;
		} else if (array[b] === array[c] && (array[b] !== null && array[c] !== null) && array[a] === null) {
			return a;
		} else if (array[a] === array[c] && (array[a] !== null && array[c] !== null) && array[b] === null) {
			return b;
		} 
	}

	if (array[4] !== aiSymbol && array[4] !== huSymbol) {
		return 4;
	}
	
	for (let i = 1; i <= 8; i++) {
		if (array[i] !== huSymbol && array[i] !== aiSymbol) {
			return i;
		}
	}
}
```